web/src/components/DandisetsPage.vue

```diff
@@ -55,48 +76,91 @@ export default {
       type: String,
       required: true,
     },
-    module: {
-      type: String,
-      required: true,
+    user: {
+      type: Boolean,
+      required: false,
+      default: false,
     },
   },
   data() {
     return {
-      selection: 'newest',
-      options: basicDandisetSortingOptions,
+      sortingOptions,
+      sortOption: Number(this.$route.query.sortOption) || 0,
+      sortDir: this.$route.query.sortDir || 1,
+      page: Number(this.$route.query.page) || 1,
     };
   },
   computed: {
-    page: {
-      get() {
-        return this.moduleRoot.page;
-      },
-      set(page) {
-        this.changePage({ page });
-      },
+    listingUrl() {
+      return this.user ? 'dandi/user' : 'dandi';
+    },
+    pages() {
+      return Math.ceil(this.totalDandisets / DANDISETS_PER_PAGE) || 1;
     },
-    moduleRoot() {
-      return this.$store.state[this.module];
+    sortField() {
+      return this.sortingOptions[this.sortOption].field;
+    },
+    queryParams() {
+      const {
+        page, sortOption, sortDir,
+      } = this;
+
+      return {
+        page,
+        sortOption,
+        sortDir,
+      };
     },
     dandisets() {
-      return this.moduleRoot.dandisets;
+      return this.dandisetRequest ? this.dandisetRequest.data : [];
     },
-    pages() {
-      return this.moduleRoot.pages;
+    totalDandisets() {
+      return this.dandisetRequest ? this.dandisetRequest.headers['girder-total-count'] : 0;
     },
   },
-  created() {
-    this.reload();
-  },
-  methods: {
-    changeSort(payload) {
-      return this.$store.dispatch(`${this.module}/changeSort`, payload);
+  asyncComputed: {
+    async dandisetRequest() {
+      const {
+        listingUrl, page, sortField, sortDir,
+      } = this;
+
+      const {
+        data, headers, status, statusText,
+      } = await girderRest.get(listingUrl, {
+        params: {
+          limit: DANDISETS_PER_PAGE,
+          offset: (page - 1) * DANDISETS_PER_PAGE,
+          sort: sortField,
+          sortdir: sortDir,
+        },
+      });
+
+      return {
+        data,
+        headers,
+        status,
+        statusText,
```

Comment:
Are these necessary? I don't see them used anywhere.

Reply:
They're not used anywhere right now, but I could see them being useful, particularly if we ever want to check if the request failed.